catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -289,15 +296,17 @@ public void setAttributeOverrides(List<String> attributeOverrides) {
             "Invalid attribute override configured for monitored directory");
       }
 
-      if (attributeOverrideMap.get(keyValue[0]) == null) {
-        attributeOverrideMap.put(keyValue[0], new ArrayList<String>());
-      }
-      ArrayList valueList = (ArrayList) attributeOverrideMap.get(keyValue[0]);
+      attributeOverrideMap.computeIfAbsent(keyValue[0], k -> new ArrayList<String>());
+      List valueList = (List) attributeOverrideMap.get(keyValue[0]);
       valueList.add(keyValue[1]);
     }
     this.attributeOverrides = attributeOverrideMap;
   }
 
+  public void setInputTransformers(List<ServiceReference<InputTransformer>> inputTransformers) {
+    this.inputTransformers = inputTransformers;
```

Comment:
❗️ Should be under lock. Any attributes you are using when locked should be updated also when under lock